### DISCUSSION TITLE
Using a class as parameter

### DISCUSSION THREAD

**Discussion by Sina-Soltani** [canton7/RestEase#202] [Q&A]
Hi,

Is it possible to use a class that includes all different properties that is needed for a method?

This is what I currently have:
```c#
public interface IMyApi
{
  [Get("{id}/{anotherId}")]
  Task<MyResponse> Get([Path] int id, [Path] int anotherId, [Query] string myQuery);
}
```

And this is what I'm looking for:
```c#
public class MyRequest
{
  [Path]
  public int Id { get; set; }

  [Path]
  public int AnotherId { get; set; }

  [Query]
  public string MyQuery { get; set; }
}

public interface IMyApi
{
  [Get("{id}/{anotherId}")]
  Task<MyResponse> Get(MyRequest request);
}
```

**Comment by canton7** [comment]
Sorry, not supported

**Reply by Sina-Soltani** [reply]
Thank you for answering.
Can it be developed?

**Comment by canton7** [comment]
Anything *can* be developed. This is quite different to how RestEase works today, so expect a reasonably long design process even if you're happy to put the effort into coding it.
